packages/mdc-auto-init/index.ts

```diff
@@ -21,16 +21,32 @@
  * THE SOFTWARE.
  */
 
-const registry = Object.create(null);
+// tslint:disable:only-arrow-functions
 
-const CONSOLE_WARN = console.warn.bind(console);
+import {MDCComponent} from '@material/base/component';
+import {MDCFoundation} from '@material/base/foundation';
+import {MDCAutoInitElement, MDCAutoInitKey} from './types';
 
-function _emit(evtType, evtData, shouldBubble = false) {
+interface ComponentClass {
+  // tslint:disable-next-line:no-any a component can pass in anything it needs to the constructor
+  new<F extends MDCFoundation>(root: Element, foundation?: F, ...args: any[]): MDCComponent<F>;
+  attachTo<F extends MDCFoundation>(root: Element): MDCComponent<F>;
+}
+
+type Registry = {
+  [K in MDCAutoInitKey]?: ComponentClass;
+};
+
+const registry: Registry = {};
+
+const CONSOLE_WARN = console.warn.bind(console); // tslint:disable-line:no-console
+
+function _emit<T extends object>(evtType: string, evtData: T, shouldBubble = false) {
```

Comment:
did we change shouldBubble --> bubble in another PR?

Reply:
Yeah, but Ken wasn't fond of the change, so I reverted it. Good eye tho! 👀

Reply:
We could rename shouldBubble across the board later in a separate PR if we want, the problem was it was renamed in base in a PR not otherwise involving base, without also updating comments across the repo that mention it as `shouldBubble`.